### DISCUSSION TITLE
[Idea]: Better persistence of store

### DISCUSSION THREAD

**Discussion by mtdvlpr** [sircharlo/meeting-media-manager#2840] [Ideas] [answered]
### Is your idea related to a problem?

The persisted store plugin can't handle date objects and defaults to persisting the whole store, which is often not what you want.

https://prazdevs.github.io/pinia-plugin-persistedstate/guide/limitations.html#non-primitive-types-are-not-persisted

### Describe your idea

https://vueuse.org/core/useLocalStorage/

The `useLocalStorage` composable automagically handles any JavaScript object and (de)serializes them to their original type.

VueUse is a collection of Vue composition utilities. It's got a lot of useful composables that clean up the code. For example:

- https://vueuse.org/core/useEventListener/
- https://vueuse.org/core/useMediaControls/
- https://vueuse.org/functions.html#category=Watch
- https://vueuse.org/functions.html#category=Array
- https://vueuse.org/core/useStepper/

### Describe alternatives you've considered

The current implementation works, but just be careful with Date objects, because they're silently transformed to strings.

**Comment by mtdvlpr** [comment]
@sircharlo, would love you opinion on this :)

**Reply by sircharlo** [reply]
I love it. I started reading up on it. Much more powerful.
I started some work on that here: d1fc75b0bc19ebe867843b0a98b973845daa5654
It's not perfect yet; there's some glitches with media lookups and whatnot, but I'm sure it just has to do with having to refactor some part of the reactive refs.

**Comment by mtdvlpr** [comment, marked as answer]
Abandoned because of too much weird migration glitches

**Reply by sircharlo** [reply]
Yeah, it's on my backburner for now, but I haven't forgotten about it. One day maybe.

**Reply by mtdvlpr** [reply]
I tried it as well, but I couldn't figure it out either. This was me abandoning it, not complaining that you might have abandoned it :)
